pkg/sql/conn_io.go

```diff
@@ -304,7 +304,12 @@ var _ Command = DeletePreparedStmt{}
 // protocol variants. So, it doesn't strictly correspond to a pgwire sync
 // message - those are not sent in the simple protocol. We synthesize Sync
 // commands though because their handling matches the simple protocol too.
-type Sync struct{}
+type Sync struct {
+	// CameFromClient specifies whether this Sync command was generated by the
+	// client during the extended protocol, or implicitly created to handle a
+	// simple protocol query.
+	CameFromClient bool
```

Comment:
I got myself confused multiple times during the review and had to come back here to re read the comment to understand what the field was for.

I think i would have been less confused if the name of the field contained its purpose as defined by the comment something like `ExplicitAndNotSkippableDuringErrorRecovery` .

Reply:
hmm i'll think of another name that might satisfy both of us. `ExplicitAndNotSkippableDuringErrorRecovery` wouldn't be my first preference, since IMHO it's a bit overly specific - it would essentially prevent this field from being used for any other purpose. more abstractly, it feels like that makes the field too strongly coupled to its usage.